### PR TITLE
force username to be lowercase

### PR DESCRIPTION
**Pull request by gnulnx**

### CODE REVIEW

File: defender/utils.py [hunk 1]
```diff
@@ -202,8 +202,8 @@ def unblock_username(username, pipe=None):
         pipe = REDIS_SERVER.pipeline()
         do_commit = True
     if username:
-        pipe.delete(get_username_attempt_cache_key(username))
-        pipe.delete(get_username_blocked_cache_key(username))
+        pipe.delete(get_username_attempt_cache_key(username.lower()))
+        pipe.delete(get_username_blocked_cache_key(username.lower()))
```
Comment (gnulnx) [inline defender/utils.py]: I wasn't sure why these lines where here twice?   

Does it make sense to leave them and then add two more that do the same with the username.lower()?

Comment (kencochrane) [inline defender/utils.py]: Sorry, I didn't see this sooner. I think it makes sense to always store usernames in the same case, lower probably. 

We could so add the .lower() inside of the ``get_username_attempt_cache_key()`` and ``get_username_blocked_cache_key()`` this way we don't have to always pass in a lower case username, and it keep it all in one place.

**Comment by kencochrane** [comment]
Should we close this, it has been open for over a year and a half with no update.

**Comment by MattBlack85** [comment]
It seems to me like a reasonable use case tough, can grab this, it doesn't look like a big amount of work

**Comment by kencochrane** [comment]
@MattBlack85 I agree, thank you!

File: defender/utils.py (hunk 1 above)
Comment (gnulnx) [inline defender/utils.py]: hey sorry guy's I guess we all dropped the ball on this one.  My apologies for not seeing this sooner as well!

**Comment by kencochrane** [comment]
closing since #90 is replacing this one. Thanks for the help.

**Pull request closed without merging**